internals/reminders_test.py

```diff
@@ -58,13 +79,31 @@ def setUp(self):
     self.current_milestone_info = {
         'earliest_beta': '2022-09-21T12:34:56',
     }
-    self.feature_template = core_models.Feature(
-      name='feature one', summary='sum', owner=['feature_owner@example.com'],
-      category=1, ot_milestone_desktop_start=100)
-    self.feature_template.key = ndb.Key('Feature', 123)
-    # This test does not require saving to the database.
+
+    self.old_feature_template = Feature(id=123,
+        name='feature one', summary='sum', owner=['feature_owner@example.com'],
+        category=1, feature_type=0, ot_milestone_desktop_start=100)
+    self.feature_template = FeatureEntry(id=123,
+      name='feature one', summary='sum',
+      owner_emails=['feature_owner@example.com'],
+      category=1, feature_type=0)
+    stages = [110, 120, 130, 140, 150, 151, 160]
+    for stage_type in stages:
+      stage = Stage(feature_id=123, stage_type=stage_type)
+      if stage_type == 150:
+        stage.milestones = MilestoneSet(desktop_first=100)
+      stage.put()
+    
+    self.old_feature_template.put()
+    self.feature_template.put()
 
     self.maxDiff = None
+  
+  def tearDown(self) -> None:
```

Comment:
I like this clean up. I wonder if we should do this more often and also do it during the setup as well.

Reply:
I agree - we've had scenarios where unrelated tests cause other tests to fail because the entities are not properly cleared out. I think this makes it a bit safer. 🙂